rules/opentitan.bzl

```diff
@@ -45,6 +46,8 @@ PER_DEVICE_DEPS = {
     "sim_dv": ["@//sw/device/lib/arch:sim_dv"],
     "fpga_cw305": ["@//sw/device/lib/arch:fpga_cw305"],
     "fpga_cw310": ["@//sw/device/lib/arch:fpga_cw310"],
+    # FIXME do we want to use different parameters or just aim for CW310 compatiblity?
+    "sim_qemu": ["@//sw/device/lib/arch:fpga_cw310"],
```

Comment:
This is not really how it work with the current bazel: in your tests you get away with using `fpga_cw310` because you only use the build artefacts. But for testing purposes, the code needs to know if the test is done using dv, verilator, qemu or real hardware to generate the proper test rules: this what the devices are for. It also means that the bazel code will literally compile one version of each binary for each device (so it will compile a QEMU version of every test). This line just says that for qemu builds, it should use the say arch lib as the cw310. So in essence it is doing exactly what you want: the QEMU target is compiling every test as if it was for the CW310.

If later we want to have a QEMU target for the cw305, then we could have `sim_qemu_cw305` and `sim_qemu_cw310` devices that would generate different `-M <machine>` options for QEMU.

I hope this makes sense :)

Reply:
I'm not sure to understand this part: " This line just says that for qemu builds, it should use the say arch lib as the cw310. So in essence it is doing exactly what you want: the QEMU target is compiling every test as if it was for the CW310."

We do not want a binary for QEMU at all, we want to use the binaries that are built FOR the FPGA_CW310 to be run on QEMU, i.e. there should be only one single binary that runs on both FPGA CW310 and on QEMU, not one for each of them. Maybe I understand your statement the wrong way; if there is no binary file generated for QEMU, that's ok.

Reply:
I don't think this is possible without a lot of hacks the current build system. But I don't think this is a problem since we can just ensure that the binary files are exactly the same as the fpga_cw310 device (this is what this line does).